pkg/prometheus/statefulset.go

```diff
@@ -327,6 +327,13 @@ func makeStatefulSetSpec(p v1alpha1.Prometheus, c *Config, ruleConfigMaps []*v1.
 			"-storage.tsdb.path=/var/prometheus/data",
 			"-storage.tsdb.retention="+p.Spec.Retention,
 		)
+
+		if len(version.Pre) > 0 && version.Pre[0].VersionStr == "alpha" {
```

Comment:
we only need to support the latest beta, if I understand correctly, we can drop this as we don't support the alphas anymore, feel free to remove them from the compatibility matrix and only keep the latest beta